crates/iota-json-rpc-types/src/iota_extended.rs

```diff
@@ -223,3 +224,19 @@ pub struct AddressMetrics {
     /// The count of daily unique sender addresses.
     pub daily_active_addresses: u64,
 }
+
+/// Provides a summary of the circulating IOTA supply.
+#[serde_as]
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct IotaCirculatingSupplySummary {
+    /// Circulating supply in IOTA at the given timestamp.
+    pub value: f64,
```

Comment:
Why is this not an u64?

Reply:
Because of the NANO to IOTA conversion

`let circulating_supply_value = circulating_supply as f64 / 1_000_000_000.0;`

See https://github.com/iotaledger/iota/blob/77a773392a41a9cd8153120dfcf5f03b36b8d51a/crates/iota-indexer/src/indexer_reader.rs#L1963

I guess we could round it down? 

Suiexplorer uses also float values https://sui-circulation.suiexplorer.com/api/current_month_sui_circulation

Reply:
> Because of the NANO to IOTA conversion

Sure, but why do this conversion in the first place and don't return the exact amount?

Reply:
using floats is a no go IMO.

Reply:
I mean, maybe it's not THAT important if information is lost by rounding in this endpoint, because it doesn't affect a single users funds, but still I wonder if we can't simply return the nanos instead.

Reply:
The idea I think is that the value is easier to understand for the integrators without another having to do another conversion. Do you have any preference here? @lzpap

The difference in return of total supply (nanos) vs circulating supply (iotas) might be however a bit problematic. Still it's a summary and could also be fine like this.

Reply:
Changed to NANOS

Reply:
I'm fine with that, but then be vocal about it in the API description, so consumers are aware that this is not in IOTAs.